### PR TITLE
build cleanups

### PR DESCRIPTION
**Pull request by Be-ing**
This simplifies the setup scripts so they only need to set VCPKG_ROOT and CMake automatically handles the rest. This way there is not a need to `source macos_buildenv.sh` every time a new shell is started; just run `cmake -D VCPKG_ROOT=buildenv/mixxx-<tab complete>`. This also makes it easy to use a local clone of the vcpkg repo instead of the prebuilt vcpkg ZIP archive which is helpful for working on vcpkg packages locally.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,22 +25,31 @@ if(POLICY CMP0071)
...
+    set(VCPKG_ROOT "$ENV{VCPKG_ROOT}")
+endif()
+
+if(DEFINED VCPKG_ROOT)
+  set(X_VCPKG_APPLOCAL_DEPS_INSTALL TRUE)
+
+  if(NOT DEFINED VCPKG_OVERLAY_PORTS)
+    set(VCPKG_OVERLAY_PORTS "${VCPKG_ROOT}/overlay/ports")
+    if(APPLE)
+        list(APPEND VCPKG_OVERLAY_PORTS "${VCPKG_ROOT}/overlay/osx")
```
Comment (daschuer) [inline CMakeLists.txt:39]: The windows branch is missing

File: CMakeLists.txt [hunk 2]
```diff
@@ -101,6 +110,10 @@ else()
   set(LLVM_CLANG false)
 endif()
 
+if(NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
```
Comment (daschuer) [inline CMakeLists.txt:113]: This can be moved into a common Apple branch together with the code below.

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you.

File: CMakeLists.txt (hunk 1 above)
Comment (Be-ing) [reply CMakeLists.txt:39]: good catch

File: CMakeLists.txt (hunk 2 above)
Comment (Be-ing) [reply CMakeLists.txt]: done

**Review by daschuer — APPROVED** [review]
LGTM, thank you.

**Pull request merged**